src/screens/charging-stations/connector-details/ChargingStationConnectorDetails.tsx

```diff
@@ -727,27 +743,93 @@ export default class ChargingStationConnectorDetails extends BaseAutoRefreshScre
-          <View style={style.rowContainer}>
-            {this.renderConnectorStatus(style)}
-            {this.renderUserInfo(style)}
-          </View>
-          <View style={style.rowContainer}>
-            {this.renderInstantPower(style)}
-            {this.renderTotalConsumption(style)}
-          </View>
-          <View style={style.rowContainer}>
-            {this.renderElapsedTime(style)}
-            {this.renderInactivity(style)}
-          </View>
-          <View style={style.rowContainer}>
-            {this.renderBatteryLevel(style)}
-            {isPricingActive ? this.renderPrice(style) : <View style={style.columnContainer} />}
-          </View>
-        </ScrollView>
+        {connector?.status === ChargePointStatus.AVAILABLE ? (
+          <ScrollView
+            contentContainerStyle={style.scrollViewContainer}
+            refreshControl={openUserModal ? null : <RefreshControl refreshing={this.state.refreshing} onRefresh={this.manualRefresh} />}>
+            <View style={style.rowContainer}>{this.renderConnectorStatus(style)}</View>
+            {isAdmin && this.renderUserSelection(style, formStyle)}
+          </ScrollView>
+        ) : (
+          <ScrollView
+            contentContainerStyle={style.scrollViewContainer}
+            refreshControl={<RefreshControl refreshing={this.state.refreshing} onRefresh={this.manualRefresh} />}>
+            <View style={style.rowContainer}>
+              {this.renderConnectorStatus(style)}
+              {this.renderUserInfo(style)}
+            </View>
+            <View style={style.rowContainer}>
+              {this.renderInstantPower(style)}
+              {this.renderTotalConsumption(style)}
+            </View>
+            <View style={style.rowContainer}>
+              {this.renderElapsedTime(style)}
+              {this.renderInactivity(style)}
+            </View>
+            <View style={style.rowContainer}>
+              {this.renderBatteryLevel(style)}
+              {isPricingActive ? this.renderPrice(style) : <View style={style.columnContainer} />}
+            </View>
+          </ScrollView>
+        )}
       </Container>
     );
   }
+
+  private openUserModal(open: boolean): void {
+    this.setState({ openUserModal: open });
+  }
+
+  private onUserSelected(users: User[]): void {
+    if (users && users.length > 0) {
+      this.setState({ selectedUser: users[0] }, () => this.openUserModal(false));
+    }
+  }
+
+  private renderUserSelection(style: any, formStyle: any) {
+    const { selectedUser } = this.state;
+    const { navigation } = this.props;
+    const userFullName = Utils.buildUserName(selectedUser);
+    return (
+      <View style={style.rowContainer}>
+        <Button block={true} style={formStyle.button} onPress={() => this.openUserModal(true)}>
+          <Text style={formStyle.buttonText} uppercase={false}>
+            {userFullName}
+          </Text>
+        </Button>
+        <Modal
```

Comment:
Put the modal in another component where I will be able to use it in the filtering framework afterwards.
The component should contain the input field + modal.
Input: the default user
Output: call back with the selected user.
You can even Abstract the input field/modal from the User with a Template class.
But we can do this is a second step.